pkg/issuer/vault/issue.go

```diff
@@ -130,7 +132,80 @@ func (v *Vault) configureCertPool(cfg *vault.Config) error {
 	return nil
 }
 
+var (
+	clientRepo   map[*v1alpha1.VaultIssuer]*vault.Client
+	clientRepoMu sync.Mutex
+)
+
+func renewToken(vaultClient *vault.Client) {
+	resp, err := vaultClient.Auth().Token().LookupSelf()
+	if err != nil {
+		glog.Warningf("vault: lookup-self failed, token renewal is disabled: %s", err)
+		return
+	}
+
+	b, _ := json.Marshal(resp.Data)
```

Comment:
Shouldn't just ignore errors here 😄

Reply:
Also, is it normal to marshal and then unmarshal the data here? This seems a bit weird 🙄 I'm not particularly experienced with the vault client library so pardon my ignorance 😄

Reply:
yes, I will fix it. marshal/unmarshal isn't really needed. I can just read the key and value straight out of resp.Data.